FRPS Server Instructions (#1474)
spectrocloud/librarium: content/docs/12-enterprise-version/07.1-reverse-proxy.md

Use the following steps to configure a reverse proxy server for Palette.

# Prerequisites

- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) is installed and available.

- [Helm](https://helm.sh/docs/intro/install/) is installed and available.

- Access to the Palette Kubernetes cluster's kubeconfig file. You can download the kubeconfig file from the Palette system console. Navigate to **Enterprise System Migration**, select the Palette cluster, and click the **Download Kubeconfig** button for the cluster.

- A domain name that you can use for the reverse proxy server. You will also need access to the DNS records for the domain so that you can create a CNAME DNS record for the reverse proxy server load balancer.

- Ensure you have an SSL certificate that matches the domain name you will assign to Spectro Proxy. You will need this to enable HTTPS encryption for the Spectro Proxy. Contact your network administrator or security team to obtain the SSL certificate. You need the following files:

>   - x509 SSL certificate file in base64 format.
> 
>   - x509 SSL certificate key file in base64 format.
> 
>   - x509 SSL certificate authority file in base64 format.

Author: ```suggestion
  - x509 SSL certificate file in base64 format

  - x509 SSL certificate key file in base64 format

  - x509 SSL certificate authority file in base64 format
```

  | **Parameter** | **Description** | **Type** |
  | --- | --- | --- |
  | `caCert`| The x509 SSL certificate authority file in base64 format. | string |
  | `clientCert`| The x509 SSL certificate file in base64 format. | string |
  | `clientKey`| The x509 SSL certificate key file in base64 format. | string |
  | `port` | The port number for the reverse proxy server. We recommend using port `443`. | integer |
  | `protocol` | The protocol to use for the reverse proxy server. We recommend using `https`. | string |
  | `server`| The domain name you will use for the Spectro Proxy server. For example, `frps.example.com`. Don't include the HTTP schema in the value. | string |

  The following is an example payload. The SSL certificate files are truncated for brevity.

  <br />

  ```json hideClipboard
    {
      "caCert": "-----BEGIN CERTIFICATE-----\n.............\n-----END CERTIFICATE-----",
      "clientCert": "-----BEGIN CERTIFICATE-----\n..........\n-----END CERTIFICATE-----",
      "clientKey": "-----BEGIN RSA PRIVATE KEY-----\n........\n-----END RSA PRIVATE KEY-----",
      "port": 443,
      "protocol": "https",
      "server": "frps.palette.example.com.com"
    }
  ```

> 

Reviewer: Let's add some space between the example payload and the infobox.

>   <InfoBox>

Author: ```suggestion
   <br />

  <InfoBox>
```

# Validate

Use the following command to validate that the Spectro Proxy server is active.

<br />

1. Open a terminal session.

2. Log in to the Palette System API by using the `/v1/auth/syslogin` endpoint. Use the `curl` command below and replace the URL with the custom domain URL you assigned to Palette or use the IP address. Ensure you replace the credentials below with your system console credentials.

  <br />

  ```bash
  curl --insecure --location 'https://palette.example.com/v1/auth/syslogin' \
   --header 'Content-Type: application/json' \
   --data '{
    "password": "**********",
    "username": "**********"
   }'
  ```
  Output
  ```json hideClipboard
  {
    "Authorization": "**********.",
    "IsPasswordReset": true
  }
  ```

3. Using the output you received, copy the authorization value to your clipboard and assign it to a shell variable. Replace the authorization value below with the value from the output.

  <br />

  ```shell hideClipboard
  TOKEN=**********
  ```

> 4. Query the system API endpoint `/v1/system/config/reverseproxy` to verify the current reverse proxy settings applied to Palette. Use the `curl` command below and replace the URL with the custom domain URL you assigned to Palette or use the IP address. You can use the `TOKEN` variable you created earlier for the authorization header.

Reviewer: ```suggestion
4. Query the system API endpoint `/v1/system/config/reverseproxy` to verify the current reverse proxy settings applied to Palette. Use the `curl` command below and replace the URL with the custom domain URL you assigned to Palette, or use the IP address. You can use the `TOKEN` variable you created earlier for the authorization header.
```

  <br />

  ```bash
  curl --location --request GET 'https://palette.example.com/v1/system/config/reverseproxy' \
   --header "Authorization: $TOKEN"
  ```

>   If the proxy server is configured correctly, you will receive an output similar to the following containing your settings. The SSL certificate outputs are truncated for brevity.

Reviewer: ```suggestion
  If the proxy server is configured correctly, you will receive an output similar to the following that contains your settings. The SSL certificate outputs are truncated for brevity.
```
